tools/replaceColumn/replaceColumn.xml

```diff
@@ -63,14 +80,26 @@ with open("output_file", 'w') as output:
                label="Which column should be replaced?" />
         <param name="skip_lines" type='integer' value='0' label="Skip this many starting lines" />
         <param name="delimiter" type="select" label="Delimited by">
-            <option value="" selected="True">Tab</option>
-            <option value=" ">Whitespace</option>
+            <option value="" selected="True">Whitespace</option>
+            <option value=" ">Tab</option>
```

Comment:
Right. It took me some figuring to understand what was going on here. So when "Tab" was selected, the resultant code that was executed was `line_content = line.split()` and also, the delimiter_local was set to `\t`.  That split is actually a whitespace split that splits on both spaces and tabs, which were then being joined back together using delimiter_local, i.e. tab characters, so any values in a file that had spaces were being split up and then getting rejoined with tabs even though the user selected to split on "Tab".  So I swapped "Tab" and "Whitespace". I had to change the behavior of the former "Whitespace" to use tabs instead, so there's no longer a "Space" option, but according to the labels, there wasn't one before anyway, so I figured no harm done.

Without the change, the last test would fail because the test file is tab-delimited but has a value in the second column with spaces.

Reply:
I could add a "Space" option to be able to output space-delimited stuff (as "Whitespace" did before).  The way this change is currently, there's no way to output space-delimited data.

I was thinking about this last night and maybe it could be a problem that "Whitespace" now (in this change) re-joins with tab instead of space. I don't know if anyone relied on that. It wasn't the default-selected option though... However the interface isn't clear on what will be used in the output because "Whitespace" is canonically either a space, tab, or any other invisible character. So what is expected in the join in the output is unclear. So perhaps, to remove the ambiguity, we could either:

* Add an additional option for the join character
* Change the "Whitespace" option to "Space" and change the behavior to split only on spaces

Incidentally, the motivation for this change was that there can be spaces in tab-delimited files, particularly tab-delimited annotation files with gene descriptions. And if a user chose to split on tab, they would get back a file that had more columns in it than in the input file.

Reply:
So what do you think?  Should I turn the 2 options into 3 options?

Before the change:

* DEFAULT Selecting "Tab" would split on whitespace (tabs and spaces) and join with tabs
* Selecting "Whitespace" would split on spaces only and join with spaces

After this PR (currently):

* DEFAULT Selecting "Whitespace" splits on whitespace (tabs and spaces) and joins with tabs
* Selecting "Tab" splits on tabs and joins with tabs

So since there's no longer a way to split & join on spaces with the current state of the PR...

Proposal:

* DEFAULT Selecting "Whitespace" splits on whitespace (tabs and spaces) and joins with tabs
* Selecting "Tab" splits on tabs and joins with tabs
* Selecting "Space" splits on spaces and joins with spaces

Reply:
OK, I implemented the proposal.  I also fixed a bug I found (in my added code).  See the last commit message for details.

Reply:
Not sure I like the idea of splitting on whitespace but joining with tabs... Though if I understand correctly, that was the original behavior, but it seems confusing at best and not likely something someone would want when simply replacing columns...

Reply:
I agree.  Not only that, but it was the default selection and was labelled "Tab".  I would have removed it, but didn't want to change the default behavior for fear it could break something else that assumed it worked that way. I had considered adding a "join delimiter" option to make what it does clear, but I'd also be willing to remove whitespace all together and deal with the potential breakages...